packages/components/calendar/src/calendar.ts

```diff
@@ -1,10 +1,10 @@
 import { buildProps, definePropType } from '@element-plus/utils/props'
 import { UPDATE_MODEL_EVENT } from '@element-plus/utils/constants'
-import type { ExtractPropTypes } from 'vue'
+import type { ExtractPropTypes, PropType } from 'vue'
 
 export const calendarProps = buildProps({
   modelValue: {
-    type: Date,
+    type: [Date, Number] as PropType<Date | number>,
```

Comment:
I don't think Calnender should accept a number, and it should always be a Date object.

our user could pass new Date(number) to Calender instead.